src/det.jl

```diff
@@ -7,25 +7,26 @@
     @inbounds return A[1]*A[4] - A[3]*A[2]
 end
 
-@inline function _det(::Size{(2,2)}, A::StaticMatrix{<:Any, <:Any, <:Unsigned})
-    @inbounds return Signed(A[1]*A[4]) - Signed(A[3]*A[2])
-end
-
 @inline function _det(::Size{(3,3)}, A::StaticMatrix)
     @inbounds x0 = SVector(A[1], A[2], A[3])
     @inbounds x1 = SVector(A[4], A[5], A[6])
     @inbounds x2 = SVector(A[7], A[8], A[9])
     return vecdot(x0, cross(x1, x2))
 end
 
-@inline function _det(::Size{(3,3)}, A::StaticMatrix{<:Any, <:Any, <:Unsigned})
-    @inbounds x0 = SVector(Signed(A[1]), Signed(A[2]), Signed(A[3]))
-    @inbounds x1 = SVector(Signed(A[4]), Signed(A[5]), Signed(A[6]))
-    @inbounds x2 = SVector(Signed(A[7]), Signed(A[8]), Signed(A[9]))
-    return vecdot(x0, cross(x1, x2))
+@inline function _det(::Size{(2,2)}, A::StaticMatrix{2, 2, <:Unsigned})
+    @inbounds return det(SMatrix{2,2,Int64}(A))
+end
+
+@inline function _det(::Size{(3,3)}, A::StaticMatrix{3, 3, <:Unsigned})
+    @inbounds return det(SMatrix{3,3,Int64}(A))
+end
+
+@inline function _det(::Size{(4,4)}, A::StaticMatrix{4, 4, <:Unsigned})
+    @inbounds return det(SMatrix{4,4,Int64}(A))
```

Comment:
`det(map(signed, A))`
is not sufficient because it might return non-sensical results since it just reinterprets 
A as signed.

I added a more general and consistent version please review.